kernel/src/context.rs

```diff
@@ -27,30 +27,29 @@ pub struct Context<'a> {
 impl Context<'_> {
     /// Gets the userspace of the current task.
     pub fn get_user_space(&self) -> CurrentUserSpace {
-        CurrentUserSpace(self.task.user_space().unwrap().vm_space().clone())
+        CurrentUserSpace::new(self.task)
     }
 }
 
 /// The user's memory space of the current task.
 ///
 /// It provides methods to read from or write to the user space efficiently.
-pub struct CurrentUserSpace(Arc<VmSpace>);
+pub struct CurrentUserSpace<'a>(&'a VmSpace);
 
-impl !Sync for CurrentUserSpace {}
-impl !Send for CurrentUserSpace {}
+/// Gets the `CurrentUserSpace` from the current task.
+///
+/// This is slower than [`Context::get_user_space`]. Don't use this getter
+/// If you get the access to the [`Context`].
+#[macro_export]
+macro_rules! get_current_userspace {
+    () => {
+        CurrentUserSpace::new(&ostd::task::Task::current().unwrap())
+    };
+}
 
-impl CurrentUserSpace {
-    /// Gets the `CurrentUserSpace` from the current task.
-    ///
-    /// This is slower than [`Context::get_user_space`]. Don't use this getter
-    /// If you get the access to the [`Context`].
-    pub fn get() -> Self {
-        let vm_space = {
-            let current_task = Task::current().unwrap();
-            let user_space = current_task.user_space().unwrap();
-            user_space.vm_space().clone()
-        };
-        Self(vm_space)
+impl<'a> CurrentUserSpace<'a> {
+    pub fn new(task: &'a Task) -> Self {
```

Comment:
Although we cannot prevent users from calling this constructor,  we should at least make it harder to misuse.

```rust
impl<'a> CurrentUserSpace<'a> {
    /// This constructor method is for internal use only.
    ///
    /// Use the `get_current_userspace` macro to get an instance of `CurrentUserSpace`.
    /// If you already have access to `Context`, call `Context::get_user_space`.
    #[doc(hidden)]
    pub fn _new(task: &'a Task) -> Self {
```

Reply:
I have refined the doc and added a debug assert to make this method hard to misuse.